dockerfiles/Dockerfile

```diff
@@ -0,0 +1,170 @@
+ARG CONTRIB_BASE_IMAGE=ghcr.io/openms/contrib:${CONTRIB_TAG}
+ARG CONTRIB_BUILD_DIR="/contrib-build"
+ARG THIRDPARTY_BRANCH="master"
+ARG THIRDPARTY_DIR="/thirdparty"
+ARG NUM_BUILD_CORES=8
+ARG MAKEFLAGS="-j${NUM_BUILD_CORES}"
+
+
+### OpenMS base ###
+FROM ${CONTRIB_BASE_IMAGE} as base
+ARG OPENMS_VERSION
+
+# the main reason we're pulling this in is to make sure that the metadata
+# below is consistently formatted for releases
+LABEL version="${OPENMS_VERSION}"
+LABEL software.version="${OPENMS_VERSION}-Ubuntu22.04"
+LABEL org.opencontainers.image.source https://github.com/OpenMS/OpenMS
+
+
+### OpenMS library ###
+FROM base as library
+ARG CONTRIB_BUILD_DIR
+ARG OPENMS_LIBRARY_DIR
+ARG OPENMS_LIBRARY_BUILD_DIR
+ARG MAKEFLAGS
+ENV MAKEFLAGS="${MAKEFLAGS}"
+
+# NOTE: this assumes that the context is root of the OpenMS repo
+COPY . ${OPENMS_LIBRARY_DIR}
+WORKDIR ${OPENMS_LIBRARY_BUILD_DIR}
+RUN /bin/bash -c "cmake -DCMAKE_BUILD_TYPE='Release' -DCMAKE_PREFIX_PATH='${CONTRIB_BUILD_DIR}/;/usr/;/usr/local' -DBOOST_USE_STATIC=OFF ${OPENMS_LIBRARY_DIR}"
+RUN make OpenMS
+LABEL software="OpenMS (library)"
+
+
+### OpenMS tools build ###
+FROM library as tools-build
+ARG OPENMS_LIBRARY_DIR
+ARG OPENMS_LIBRARY_BUILD_DIR
+ARG MAKEFLAGS
+ARG THIRDPARTY_BRANCH
+ARG THIRDPARTY_DIR
+ENV MAKEFLAGS="${MAKEFLAGS}"
+
+RUN apt-get update && apt-get install -y --no-install-recommends --no-install-suggests openjdk-17-jdk
+
+WORKDIR ${OPENMS_LIBRARY_DIR}
+RUN <<-EOF
+    mkdir /thirdparty
+    # note, we are not using the submodule here because the .git folder
+    # is not copied to this stage. Instead, we clone the repo directly.
+    git clone --branch=${THIRDPARTY_BRANCH} --depth=1 https://github.com/OpenMS/THIRDPARTY.git
+
+    # copying only the relevant binaries and leaving the rest behind
+    cp -r THIRDPARTY/All/* ${THIRDPARTY_DIR}
+    cp -r THIRDPARTY/Linux/64bit/* ${THIRDPARTY_DIR}
+
+    rm -rf THIRDPARTY
+EOF
+
+ENV PATH="${THIRDPARTY_DIR}/LuciPHOr2:${THIRDPARTY_DIR}/MSGFPlus:${THIRDPARTY_DIR}/Sirius:${THIRDPARTY_DIR}/ThermoRawFileParser:${THIRDPARTY_DIR}/Comet:${THIRDPARTY_DIR}/Fido:${THIRDPARTY_DIR}/MaRaCluster:${THIRDPARTY_DIR}/Percolator:${THIRDPARTY_DIR}/SpectraST:${THIRDPARTY_DIR}/XTandem:${THIRDPARTY_DIR}/Sage:${PATH}"
+WORKDIR ${OPENMS_LIBRARY_BUILD_DIR}
+# RUN make TOPP && rm -rf src doc CMakeFiles
+RUN make TOPP
+WORKDIR /
+ENV PATH="${OPENMS_LIBRARY_BUILD_DIR}/bin:${PATH}"
+
+### OpenMS tools runtime ###
+FROM ubuntu:22.04 AS tools
+ARG OPENMS_LIBRARY_DIR
+ARG OPENMS_LIBRARY_BUILD_DIR
+ARG CONTRIB_BUILD_DIR
+ARG OPENMS_LIBRARY_DIR
+ARG THIRDPARTY_DIR
+
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends --no-install-suggests \
+    libqt5opengl5 \
+    libsvm3 \
+    libzip4 \
+    zlib1g \
+    libbz2-1.0 \
+    libgomp1 \
+    libqt5svg5 \
+    libxerces-c3.2 \
+    libboost-date-time1.74-dev  \
+    libboost-iostreams1.74-dev \
+    libboost-regex1.74-dev \
+    libboost-math1.74-dev \
+    libboost-random1.74-dev \
```

Comment:
I'm kind of missing an ILP solver here. CoinOR (preferably due to license) or GLPK.
I wonder how this successfully configures.

Reply:
Also for full functionality you might want `mono/.NET` for ThermoRawFileParser. But it's  a beast so I am not sure.

Reply:
Hmm, good point, I will check out the CoinOR issue -- is there a test that should catch this?

Regarding `mono` or `.NET`, I lean towards not including it unless some OpenMS component depends on it directly, or maybe if you included it in your THIRDPARTY repository.

Reply:
Regarding CoinOR: LPWrapper_test or something should usually fail. Actually the CMake configuration stage that looks for dependencies should already fail. So I assume it is finding it somewhere. Are you currently testing with the contrib image as a base image maybe? In that case none of the apt dependencies are needed of course. Contrib and apt dependencies are mutually exclusive.

Regarding mono: We could actually also distinguish between openms-tools and openms-thirdparty as we do on conda.
Where only openms-thirdparty contains THIRDPARTY and its dependencies.
I just mean: If we do not install mono for a lightweight image, it does not make sense to pull the ThermoRawFileParser libs and bins from our THIRDPARTY repo, as they will be unusable on the image in that state.

Reply:
Yes correct. It would be cool if during the transition phase we could choose between a build that copies from the contrib container or one that downloads from apt.
But mid-term apt is easier to maintain.

Reply:
Maybe for openms-lib we can run class tests and for openms-tools the TOPP tests.
But for now TOPP tool tests should cover most of it.

Reply:
> It would be cool if during the transition phase we could choose between a build that copies from the contrib container or one that downloads from apt.

If we merge https://github.com/OpenMS/contrib/pull/136 first, then we can just go with apt here, no? I don't think anything should break. I'd rather avoid that if you agree.

> Maybe for openms-lib we can run class tests and for openms-tools the TOPP tests. But for now TOPP tool tests should cover most of it.

Okay, I think that running the class tests for the library makes sense. You have the tests, we might as well make use of them :)